staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -137,8 +143,13 @@ func (t *envelopeTransformer) TransformFromStorage(ctx context.Context, data []b
 		return nil, false, err
 	}
 
+	encryptedObjectCacheKey, err := generateCacheKey(encryptedObject.EncryptedDEK, encryptedObject.KeyID, encryptedObject.Annotations)
+	if err != nil {
+		return nil, false, err
+	}
+
 	// Look up the decrypted DEK from cache first
-	transformer := t.cache.get(encryptedObject.EncryptedDEK)
+	transformer := t.cache.get(encryptedObjectCacheKey)
```

Comment:
The call below to `addTransformerForDecryption` is incorrect - please add a test that catches that :)

Reply:
good catch! interesting how the current tests didn't catch it. Let me check!

Reply:
Added a new test in `TestEnvelopeCaching` and modified it to do multiple reads to test this. The new table test failed as expected to catch this

```bash
=== RUN   TestEnvelopeCaching
=== RUN   TestEnvelopeCaching/entry_in_cache_should_withstand_plugin_failure
=== RUN   TestEnvelopeCaching/cache_entry_expired_should_not_withstand_plugin_failure
=== RUN   TestEnvelopeCaching/cache_entry_expired_should_work_after_cache_refresh
    /home/anramase/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope_test.go:221: expected 1 decrypt calls, got 10
--- FAIL: TestEnvelopeCaching (0.00s)
    --- PASS: TestEnvelopeCaching/entry_in_cache_should_withstand_plugin_failure (0.00s)
    --- PASS: TestEnvelopeCaching/cache_entry_expired_should_not_withstand_plugin_failure (0.00s)
    --- FAIL: TestEnvelopeCaching/cache_entry_expired_should_work_after_cache_refresh (0.00s)
FAIL
FAIL	k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2	0.005s
```